src/compiler/utilities.ts

```diff
@@ -6258,6 +6258,34 @@ namespace ts {
         return isSourceFile(node) || isModuleBlock(node) || isBlock(node) && isFunctionLike(node.parent);
     }
 
+    /* @internal */
+    export function isNewThisScope(node: Node): boolean {

```

Comment:
It's probably worth describing what this function does. Maybe you could even inline it.